com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/AtlasBuilder.java

```diff
@@ -67,16 +64,12 @@ private static int getPageCount(List<BufferedImage> images, TextureImage.Type te
     }
 
     @Override
-    public Task<TextureImage.Type> create(IResource input) throws IOException, CompileExceptionError {
-        Atlas.Builder builder = Atlas.newBuilder();
-        ProtoUtil.merge(input, builder);
+    public Task create(IResource input) throws IOException, CompileExceptionError {
+        Atlas.Builder builder = getMessageBuilder(input);
         Atlas atlas = builder.build();
 
-        TextureImage.Type textureImageType = getTexureType(atlas);
```

Comment:
it seems like there is no reason to know TexureType until we build task (We don't use it in `create()`)